### PR TITLE
improve DisplayHelper to handle frame rate changes

### PR DESCRIPTION
**Pull request by pixelflinger**

### CODE REVIEW

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 1]
```diff
@@ -16,42 +16,163 @@
...
+            listener.onDisplayChanged();
+        }
     }
 
-    public static float getRefreshRate(Display display) {
-        return display.getRefreshRate();
+    /**
+     * Unsets the {@link DisplayListener} if any
+     */
+    public void unsetListener() {
```
Comment (romainguy) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:105]: On Android we use either addListener/removeListener() or setListener() but we accept null to unset it.

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 2]
```diff
@@ -16,42 +16,163 @@
...
+            @Override
+            public void onDisplayAdded(int displayId) {
+            }
+            @Override
+            public void onDisplayRemoved(int displayId) {
+            }
+            @Override
+            public void onDisplayChanged(int displayId) {
+                if (displayId == display.getDisplayId()) {
+                    listener.onDisplayChanged();
```
Comment (romainguy) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:83]: Shouldn't you use `handler` here when `handler` is not null?

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 3]
```diff
@@ -16,42 +16,163 @@
...
+     * Sets a callback to by notified when a given {@link Display} properties change.
+     *
+     * The callback is always automatically called once when this function is called. There can
+     * only be a single {@link DisplayListener} set.
+     *
+     * @param listener  the callback to invoke
+     * @param display   the {@link Display} to monitor changes for
+     * @param handler   an optional {@link Handler} to invoke the callback on
+     */
+    public void setListener(@NonNull DisplayListener listener, @NonNull Display display, Handler handler) {
```
Comment (romainguy) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:71]: Use `@Nullable`on optional parameters

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 4]
```diff
@@ -16,42 +16,163 @@
...
+    /**
+     * Populate a {@link Renderer.DisplayInfo} with properties from the given {@link Display}
+     *
+     * @param display   {@link Display} to get {@link Renderer.DisplayInfo} from
+     * @param info      an instance of {@link Renderer.DisplayInfo} or null
+     * @return          an populated instance of {@link Renderer.DisplayInfo}
+     */
     @NonNull
-    public static Renderer.DisplayInfo getDisplayInfo(Display display, @NonNull Renderer.DisplayInfo info) {
+    public static Renderer.DisplayInfo getDisplayInfo(@NonNull Display display, Renderer.DisplayInfo info) {
```
Comment (romainguy) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:129]: @Nullable on the info parameter

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 5]
```diff
@@ -16,42 +16,189 @@
...
+    private Display mDisplay;
+    private Renderer mRenderer;
+    private DisplayManager.DisplayListener mListener;
+
+    /**
+     * Creates a DisplayHelper which helps managing a {@link Display}.
+     *
+     * The {@link Display} to manage is specified with {@link #attach}
+     *
+     * @param context a {@link Context} to used to retrieve teh {@link DisplayManager}
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:47]:
```suggestion
     * @param context a {@link Context} to used to retrieve the {@link DisplayManager}
```

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 6]
```diff
@@ -16,42 +16,189 @@
...
+     */
+    public DisplayHelper(@NonNull Context context) {
+        mDisplayManager = (DisplayManager) context.getSystemService(Context.DISPLAY_SERVICE);
+    }
+
+    /**
+     * Creates a DisplayHelper which helps managing a {@link Display} and provides a Handler
+     * where callbacks can execute filament code. Use this method if filament is executing
+     * on another thread.
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:56]:
```suggestion
     * Creates a DisplayHelper which helps manage a {@link Display} and provides a Handler
     * where callbacks can execute filament code. Use this method if filament is executing
     * on another thread.
```

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 7]
```diff
@@ -16,42 +16,189 @@
...
-        return 0;
     }
 
-    public static long getPresentationDeadlineNanos(Display display) {
-        if (Build.VERSION.SDK_INT >= 29) {
-            return display.getPresentationDeadlineNanos();
+    /**
+     * Sets the filament {@link Renderer} associated to the {@link Display}, from this point
+     * on, {@link Renderer.DisplayInfo} will be automatically updated when the {@link Display}
+     * properties changes.
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:79]:
```suggestion
     * properties change.
```

File: android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java [hunk 8]
```diff
@@ -16,42 +16,189 @@
...
+     */
+    public static long getAppVsyncOffsetNanos(@NonNull Display display) {
+        if (Build.VERSION.SDK_INT >= 29) {
+            return display.getAppVsyncOffsetNanos();
+        }
+        return 0;
+    }
+
+    /**
+     * @return the {@link Display} presentation deadline before the h/w vsync event is nanosecond
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/android/DisplayHelper.java:177]:
```suggestion
     * @return the {@link Display} presentation deadline before the h/w vsync event in nanoseconds
```

**Pull request merged**